src/transformers/utils/generic.py

```diff
@@ -1015,6 +1015,8 @@ def wrapper(self, *args, **kwargs):
                 all_args[k] = v
 
         capture_flags = _CAN_RECORD_REGISTRY.get(str(self.__class__), {})  # there is a weak ref for executorch
+        capture_flags_hidden_states = (capture_flags.get("hidden_states", ()),)
```

Comment:
Note that with this code, if `"hidden_states"` is not found, we get: `capture_flags_hidden_states = ((),)`. I tried this with `isinstance` and it works. Apparently, this is because Python checks for tuples and visits them recursively, so I guess we can leave it as is, but it can be confusing.